autowiring/auto_signal.h

```diff
@@ -44,10 +61,24 @@ namespace autowiring {
       signal_node(std::function<void(Args...)>&& fn) :
         fn(std::move(fn))
       {}
-
+      
+      //Functions where the first argument is a signal_node<...> or base type are also ok.
+      signal_node(std::function<void(signal_node<Args...>*, Args...)>&& newFn) :
+      fn([this, newFn](Args... args){ newFn(this, args...); })
```

Comment:
Nice rewrite.

In this case, recommend you consider adding another `std::function` as a member variable rather than capturing `newFn` in your closure.  If you do that, then you can use `std::move` to transfer `newFn` to your member variable, potentially making it a bit faster.  Also it's a little clearer.